hooks/tk-multi-publish2/ingest/ingest_files.py

```diff
@@ -0,0 +1,198 @@
+    """
+    Inherits from PublishFilesPlugin
+    """
+
+    @property
+    def description(self):
+        """
+        Verbose, multi-line description of what the plugin does. This can
+        contain simple html for formatting.
+        """
+
+        desc = super(IngestFilesPlugin, self).description
+
+        return desc + "<br><br>" + """
+        After Ingesting the file a publishedFile gets created. The publishedFile will also be linked to a Plate entity.
+        """
+
+    def publish(self, task_settings, item):
+        """
+        Executes the publish logic for the given item and task_settings.
+
+        :param task_settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the task_settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        """
+
+        super(IngestFilesPlugin, self).publish(task_settings, item)
+
+        # create a plate entity after the publish has went through successfully.
+        plate_entity = self._create_plate_entity(item)
+
+        # let's create ingest_plate_data within item properties,
+        # so that we can link the version created to plate entity as well.
+        item.properties["ingest_plate_data"] = plate_entity
+
+        # link the publish file to our plate entity.
+        self._link_published_files_to_plate_entity(plate_entity, item)
+        self.logger.info("Plate entity registered and Publish file is linked!")
+
+    def finalize(self, task_settings, item):
+        """
+        Execute the finalization pass. This pass executes once
+        all the publish tasks have completed, and can for example
+        be used to version up files.
+
+        :param task_settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the task_settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        """
+
+        super(IngestFilesPlugin, self).finalize(task_settings, item)
+
+        # get the data for the plate that was just created in SG
+        plate_data = item.properties["ingest_plate_data"]
+
+        path = item.properties["path"]
+
+        self.logger.info(
+            "Plate created for file: %s" % (path,),
+            extra={
+                "action_show_in_shotgun": {
+                    "label": "Show Plate",
+                    "tooltip": "Open the Publish in Shotgun.",
+                    "entity": plate_data
+                }
+            }
+        )
+
+    @staticmethod
+    def _validate_plate_entity(item):
+        sg = get_sg_connection()
+        sg_filters = [
+            ['project', 'is', item.context.project],
+            ['code', 'is', item.properties["publish_name"]]
+        ]
+        if item.context.entity:
+            if item.context.entity["type"] == "Shot":
+                sg_filters.append(['shots', 'is', item.context.entity])
```

Comment:
Use `sg_shot` not `shots`

Reply:
Any reason we are not updating this field? We were updating this in plate_ingest.

Reply:
`shots` should/will be used for specifying instances of that plate, not the plate entity itself